### PR TITLE
Allow unsaved entity to be sent to the API

### PR DESCRIPTION
**Pull request by gandarez**
This PR allows any unsaved/empty entity/extra heartbeats to be sent to the API by setting the flag `is-unsaved-entity` to **true**. The following validations will be skipped:
* formatting
* remote detection
* dependency detection

Those ones will always run for every single heartbeat:
* entity filtering (with exceptions)
* entity modifier
* language detection
* project detection (with exceptions)
* project filtering
* sanitization

Feature to solve [this issue](https://github.com/wakatime/vscode-wakatime/issues/266).

### CODE REVIEW

File: cmd/heartbeat/heartbeat.go [hunk 1]
```diff
@@ -193,42 +193,60 @@ func buildHeartbeats(params paramscmd.Params) []heartbeat.Heartbeat {
...
+		opts = append(opts,
+			remote.WithDetection(),
+			filestats.WithDetection(filestats.Config{
+				LinesInFile: params.Heartbeat.LinesInFile,
+			}),
+			language.WithDetection(),
+			deps.WithDetection(deps.Config{
+				FilePatterns: params.Heartbeat.Sanitize.HideFileNames,
+			}),
+			project.WithDetection(project.Config{
```
Comment (alanhamlett) [inline cmd/heartbeat/heartbeat.go]: Skipping project detection means the value from `--project` and `--alternate-project` aren't used?

File: cmd/root.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ func NewRootCMD() *cobra.Command {
 
 func setFlags(cmd *cobra.Command, v *viper.Viper) {
 	flags := cmd.Flags()
+	flags.Bool("allow-unsaved-entity", false, "Skip checking for file existing and any detection on the entity.")
```
Comment (alanhamlett) [inline cmd/root.go]: Let's use description:

`Normally files that don't exist on disk are skipped and not tracked. When this option is present, the file will be tracked even if it doesn't exist.`

**Review by alanhamlett — CHANGES_REQUESTED** [review]


File: cmd/heartbeat/heartbeat.go (hunk 1 above)
Comment (gandarez) [reply cmd/heartbeat/heartbeat.go]: Language and project detection can run under circunstances. I'm fixing it in this PR.

**Review by alanhamlett — CHANGES_REQUESTED** [review]
I just realized, we need `config.AllowUnsavedEntity` tied to the heartbeat, because we don't want to prevent extra heartbeats that are real files to skip project and language detection.

**Comment by gandarez** [comment]
> I just realized, we need `config.AllowUnsavedEntity` tied to the heartbeat, because we don't want to prevent extra heartbeats that are real files to skip project and language detection.

Do you mean a new attribute for extra heartbeats to individually control each one or only the root param is enough?

**Review by alanhamlett — APPROVED** [review]

**Pull request merged**